platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/reconnect/LearnerPullVirtualTreeView.java

```diff
@@ -215,20 +224,27 @@ public void readNode(final SerializableDataInputStream in, final long path, fina
         assert !firstNodeResponse : "Root node must be the first node received from the teacher";
         final boolean isLeaf = isLeaf(path);
         traversalOrder.nodeReceived(path, isClean);
+
         if (isLeaf) {
             if (firstLeaf) {
                 root.prepareForFirstLeaf();
                 firstLeaf = false;
             }
             if (!isClean) {
                 final VirtualLeafRecord<K, V> leaf = in.readSerializable(false, VirtualLeafRecord::new);
+                mapStats.incrementLeafData(1, 0);
                 assert path == leaf.getPath();
                 nodeRemover.newLeafNode(path, leaf.getKey());
                 root.handleReconnectLeaf(leaf); // may block if hashing is slower than ingest
             }
         }
     }
 
+    /** Returns the ReconnectMapStats object. */
+    public ReconnectMapStats getMapStats() {
```

Comment:
missing return tag, and nullity annotation

Reply:
added